From 58b36ceaeb6dcdbcee54d5849b282dd10f9b3f24 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Fri, 16 May 2014 14:15:04 +0200
Subject: [PATCH 1/2] 730243 Consolidate GErrorSafePtr definitions

       * src/common/nmv-error.h: New file.
       * src/common/nmv-error.cc: Likewise.
       * src/common/Makefile.am: Add nmv-error.[h|cc] to the build
       system.
       * src/common/nmv-ustring.cc: Remove local copy of GErrorSafePtr.
       (wstring_to_ustring): Use GErrorSafePtr from nmv-error.h
       instead of local copy.
       (ustring_to_wstring): Likewise.
       * src/confmgr/nmv-gconf-mgr.cc: Remove local copy of
       GErrorSafePtr.
---
 src/common/Makefile.am       |  6 +++--
 src/common/nmv-error.cc      | 51 +++++++++++++++++++++++++++++++++++++++++++
 src/common/nmv-error.h       | 52 ++++++++++++++++++++++++++++++++++++++++++++
 src/common/nmv-ustring.cc    | 20 +++--------------
 src/confmgr/nmv-gconf-mgr.cc | 11 +---------
 5 files changed, 111 insertions(+), 29 deletions(-)
 create mode 100644 src/common/nmv-error.cc
 create mode 100644 src/common/nmv-error.h

diff --git a/src/common/Makefile.am b/src/common/Makefile.am
index 074c5bd..9ef837a 100644
--- src/common/Makefile.am
+++ src/common/Makefile.am
@@ -42,7 +42,8 @@ $(h)/nmv-insert-statement.h \
 $(h)/nmv-delete-statement.h \
 $(h)/nmv-proc-utils.h \
 $(h)/nmv-proc-mgr.h \
-$(h)/nmv-loc.h
+$(h)/nmv-loc.h \
+$(h)/nmv-error.h
 
 libnemivercommon_la_SOURCES= $(headers) \
 $(h)/nmv-ustring.cc \
@@ -71,7 +72,8 @@ $(h)/nmv-sql-statement.cc \
 $(h)/nmv-insert-statement.cc \
 $(h)/nmv-delete-statement.cc \
 $(h)/nmv-proc-utils.cc \
-$(h)/nmv-proc-mgr.cc
+$(h)/nmv-proc-mgr.cc \
+$(h)/nmv-error.cc
 
 publicheaders_DATA=$(headers)
 publicheadersdir=$(NEMIVER_INCLUDE_DIR)/common
diff --git a/src/common/nmv-error.cc b/src/common/nmv-error.cc
new file mode 100644
index 0000000..aa7a2be
--- /dev/null
+++ src/common/nmv-error.cc
@@ -0,0 +1,51 @@
+/* -*- Mode: C++; indent-tabs-mode:nil; c-basic-offset: 4-*- */
+
+/*Copyright (c) 2014 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies
+ * or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS",
+ * WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE
+ * AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+#include "config.h"
+#include "nmv-error.h"
+
+NEMIVER_BEGIN_NAMESPACE (nemiver)
+NEMIVER_BEGIN_NAMESPACE (common)
+
+void
+GErrorRef::operator () (GError *a_error)
+{
+}
+
+void
+GErrorUnref::operator () (GError *a_error)
+{
+    if (a_error) {
+        g_error_free (a_error);
+    }
+}
+
+NEMIVER_END_NAMESPACE (common)
+NEMIVER_END_NAMESPACE (nemiver)
+
diff --git a/src/common/nmv-error.h b/src/common/nmv-error.h
new file mode 100644
index 0000000..0230087
--- /dev/null
+++ src/common/nmv-error.h
@@ -0,0 +1,52 @@
+/* -*- Mode: C++; indent-tabs-mode:nil; c-basic-offset: 4-*- */
+
+/*Copyright (c) 2014 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies
+ * or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS",
+ * WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE
+ * AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+#ifndef __NMV_ERROR_H__
+#define __NMV_ERROR_H__
+
+#include <glib.h>
+#include "nmv-namespace.h"
+#include "nmv-safe-ptr.h"
+
+namespace nemiver {
+namespace common {
+
+struct GErrorRef {
+    void operator () (GError *a_error);
+};
+
+struct GErrorUnref {
+    void operator () (GError *a_error);
+};
+
+typedef SafePtr<GError, GErrorRef, GErrorUnref> GErrorSafePtr;
+
+}//end namespace common
+}//end namespace nemiver
+#endif // __NMV_ERROR_H__
diff --git a/src/common/nmv-ustring.cc b/src/common/nmv-ustring.cc
index f5dfe49..076258c 100644
--- src/common/nmv-ustring.cc
+++ src/common/nmv-ustring.cc
@@ -30,6 +30,7 @@
 #include "config.h"
 #include <cstring>
 #include <sstream>
+#include "nmv-error.h"
 #include "nmv-ustring.h"
 #include "nmv-safe-ptr-utils.h"
 #include "nmv-log-stream-utils.h"
@@ -479,21 +480,6 @@ WString::assign (super_type::size_type a_n, gunichar a_c)
     return *this;
 }
 
-struct GErrorRef {
-    void operator () (GError *)
-    {
-    }
-};
-
-struct GErrorUnref {
-    void operator () (GError *a_err)
-    {
-        if (a_err) {
-            g_error_free (a_err);
-        }
-    }
-};
-
 bool
 wstring_to_ustring (const WString &a_wstr,
                     UString &a_ustr)
@@ -504,7 +490,7 @@ wstring_to_ustring (const WString &a_wstr,
     utf8_buf.reset (g_ucs4_to_utf8 (a_wstr.c_str (),
                                     a_wstr.size (), &wstr_len,
                                     &utf8_bytes_len, &err));
-    SafePtr<GError, GErrorRef, GErrorUnref> error;
+    GErrorSafePtr error;
     error.reset (err);
     if (error) {
         LOG_ERROR ("got error conversion error: '" << error->message << "'");
@@ -531,7 +517,7 @@ ustring_to_wstring (const UString &a_ustr,
                                 &utf8_bytes_len,
                                 &wstr_len,
                                 &err));
-    SafePtr<GError, GErrorRef, GErrorUnref> error;
+    GErrorSafePtr error;
     error.reset (err);
     if (error) {
         LOG_ERROR ("got error conversion error: '" << error->message << "'");
diff --git a/src/confmgr/nmv-gconf-mgr.cc b/src/confmgr/nmv-gconf-mgr.cc
index fd7ff1c..f0f97ad 100644
--- src/confmgr/nmv-gconf-mgr.cc
+++ src/confmgr/nmv-gconf-mgr.cc
@@ -24,6 +24,7 @@
  */
 #include <gconf/gconf-client.h>
 #include "nmv-i-conf-mgr.h"
+#include "common/nmv-error.h"
 #include "common/nmv-ustring.h"
 #include "common/nmv-exception.h"
 #include "common/nmv-safe-ptr-utils.h"
@@ -92,16 +93,6 @@ public:
 
 //static const char * NEMIVER_KEY_DIR = "/app/nemiver";
 
-struct GErrorRef {
-    void operator () (GError *a_error) {if (a_error) {}}
-};
-
-struct GErrorUnref {
-    void operator () (GError *a_error) {if (a_error) {g_error_free (a_error);}}
-};
-
-typedef SafePtr<GError, GErrorRef, GErrorUnref> GErrorSafePtr;
-
 void
 client_notify_func (GConfClient *a_client,
                     const char* a_key,
-- 
1.9.0

